enhancements/profile-versioning.md

```diff
@@ -0,0 +1,449 @@
+versioned using `Revision $INT`, which doesn't meet strict semantic versioning
+guidance).
+
+### Implementation Details/Notes/Constraints [optional]
+
+There are two considerations for how versions interact with profiles.
+
+The first is if there is only one profile, and it has a single version. For
+example, supporting the CIS OpenShift 1.4.0 benchmark. The second is if there
+is a single profile, but multiple versions. For example, PCI-DSS version 3.2.1
+and version 4.0.0.
+
+#### Single Profile & Single Version
+
+This case can be implemented in the following steps:
+
+1. Associate a version to the profile in
+   [ComplianceAsCode/content](https://github.com/ComplianceAsCode/content)
+   using the `metadata.version` attribute of the profile
+2. Update the operator's profile parser to parse the version from the data
+   stream, and set the appropriate attribute on the `Profile` custom resource
+
+This will look similar to the following after the profiles are parsed by the
+operator on start-up:
+
+```console
+$ oc get profiles.compliance
+NAME                             AGE   VERSION
+ocp4-cis                         21s   1.4.0
+ocp4-cis-node                    22s   1.4.0
+ocp4-pci-dss                     21s   3.2.1
+ocp4-pci-dss-node                21s   3.2.1
+```
+
+Profiles in the
+[ComplianceAsCode/content](https://github.com/ComplianceAsCode/content) that do
+not have a `metadata.version`, will not have that attribute set in the
+`Profile` custom resource (using `ocp4-nerc-cip` purely as an example below):
+
+```console
+$ oc get profiles.compliance
+NAME                             AGE   VERSION
+ocp4-cis                         21s   1.4.0
+ocp4-cis-node                    22s   1.4.0
+ocp4-e8                          21s
+ocp4-pci-dss                     21s   3.2.1
+ocp4-pci-dss-node                21s   3.2.1
+rhcos4-e8                        11s
+```
+
+#### Single Profile & Multiple Versions
+
+The operator must account for cases where a profile has multiple versions.
+
+Due to limitations of the data stream, each profile can only have one version.
+Additionally, there can only be one profile using a given name in the data
+stream.
+
+The operator can expose multiple versions for a single profile using the following approach:
+
+1. Version each profile to the initially supported version, keeping the profile name the same
+2. When a new version is released, implement it using the versionless profile named
+3. Implement a new profile for the older version under a new name
+
+For example, this would look like the following in practice:
+
+1. Version the current CIS OpenShift benchmark at version 1.4.0
+2. When version 1.5.0 is implemented, move version 1.4.0 to a new profile named `ocp4-cis-1.4.0`)
+3. Create a new profile named `ocp4-cis-1.5.0`
+4. Release the operator with `ocp4-cis` as version 1.5.0, so that the versionless name always references the latest version
```

Comment:
So if a user has a scheduled scan with ocp4-cis(-1.4.0), should the content image being used be updated to newer versions ocp4-cis(-1.5.0) automatically when ocp4-cis(-1.5.0) is released? Thanks.

Reply:
> So if a user has a scheduled scan with ocp4-cis(-1.4.0), should the content image being used be updated to newer versions ocp4-cis(-1.5.0) automatically when ocp4-cis(-1.5.0) is released? Thanks.

Yep - this will match the default behavior that exists today.

Reply:
Additionally - if a users wishes to pin to a specific version they can use `ocp4-cis-1.4.0` as the profile name in the binding.